### PR TITLE
Enforce better naming convention

### PR DESCRIPTION
**Pull request by isBatak**
Solves #42 and #44

What's the best way to test these changes?
This issue https://github.com/infinum/js-linters/discussions/42#discussioncomment-1640261 may still exist.

Todo
- [x] add test after this is merged https://github.com/infinum/js-linters/pull/80

### CODE REVIEW

**Comment by fvoska** [comment]
To test the changes, it is best to create an example project that uses the config and write some code that would trigger linting errors/warnings. That is how we did it for Angular-specific eslint config [here](https://github.com/infinum/js-linters/tree/master/packages/eslint-config-angular-ts) and we could probably do the same for the base TS config with some simple TS example project

**Comment by fvoska** [comment]
As for the issue from the comment https://github.com/infinum/js-linters/discussions/42#discussioncomment-1640261, I had the same issue on some angular repo where I added some 3rd party rule that require TS compiler and there I also had to define `parserOptions.project`. I think this is something that will have to be done in each project individually because the tsconfig file that we have to point to might be in a different location or named differently.

We should update the readme with this info.

File: examples/angular/package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "@angular-eslint/template-parser": "12.5.0",
     "@angular/cli": "~12.2.10",
     "@angular/compiler-cli": "~12.2.0",
-    "@infinumjs/eslint-config-angular-ts": "file:..",
+    "@infinumjs/eslint-config-angular-ts": "@infinumjs/eslint-config-angular-ts",
```
Comment (fvoska) [inline examples/angular/package.json:35]: I didn't know about this way to define a version. Is that something specific to how lerna does linking?

Comment (isBatak) [reply examples/angular/package.json:35]: Yes, that's the whole point of monorepo. 
Now, when we have two folders that are in monorepo setup (`packages`, `examples`) we can share packages between them. 
Lerna will symlink `@infinumjs/eslint-config-angular-ts` package source into Angular project.

Comment (isBatak) [reply examples/angular/package.json:35]: This is mistake, we should set a version here.

**Comment by kristijanbambir** [comment]
Is this PR done or is there more work to be done? If it's not done, please convert it to a draft until completed.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Contributing
+
+## Project setup
+
+1. Install root packages `npm install`
```
Comment (kristijanbambir) [inline CONTRIBUTING.md:5]:
```suggestion
1. Install root packages with `npm install`
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+2. Run `npm run bootstrap`
```
Comment (kristijanbambir) [inline CONTRIBUTING.md:6]:
```suggestion
2. Run `npm run bootstrap` to symlink dependencies
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+## Testing examples
+
+1. Navigate to specific example project (angular or nextjs) `cd examples/angular`
```
Comment (kristijanbambir) [inline CONTRIBUTING.md:10]:
```suggestion
1. Navigate to specific a example project (Angular or Next.js)
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+2. Open it in new VSCode instance `code .`
```
Comment (kristijanbambir) [inline CONTRIBUTING.md:11]: Why does it need to be opened in VSCode?

**Review by kristijanbambir — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+3. Run lint command `npm run lint`
```
Comment (kristijanbambir) [inline CONTRIBUTING.md:12]:
```suggestion
3. Run the lint command with `npm run lint`
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+4. Optional run test command (only nextjs) `npm run test`
```
Comment (kristijanbambir) [inline CONTRIBUTING.md:13]:
```suggestion
4. Optionally run the test command (only for Next.js) with `npm run test`
```

**Review by kristijanbambir — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 5 above)
Comment (isBatak) [reply CONTRIBUTING.md:11]: Should we be more generic and write something like "Open it with you default code editor, e.g. `code .`

Comment (kristijanbambir) [reply CONTRIBUTING.md:11]: I would omit this step completely, since it has nothing to do with the step that comes after.

Comment (isBatak) [reply CONTRIBUTING.md:11]: it does! If you run lint inside monorepo some weird things are happening. You told me that we should test linter rules bu opening them in separate VSCode instance outside monorepo.

Comment (kristijanbambir) [reply CONTRIBUTING.md:11]: I think you are assuming that one needs to open VSCode and use its terminal to run commands. I would not mention VSCode at all. These steps are purely about running commands in any terminal.

The issue we talked about it is  about the ESLint plugin for VSCode that doesn't work in monorepos properly.

**Comment by adrianbicanic** [comment]
@isBatak What updates or info related to this PR?

**Comment by isBatak** [comment]
@adrianbicanic I have to revisit this PR and add tests

**Review by kristijanbambir — APPROVED** [review]

**Comment by kamdubiel** [comment]
Done in:
https://github.com/infinum/js-linters/pull/128

**Pull request closed without merging**